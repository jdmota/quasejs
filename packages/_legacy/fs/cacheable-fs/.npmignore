*.txt
__dev__
test
node_modules
yarn.lock
