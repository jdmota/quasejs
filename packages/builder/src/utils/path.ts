const path = require( "path" );
const slash = require( "slash" );

export const reExt = /\.(.+)$/;

export function lowerPath( p: string ) {
  return path.sep === "\\" ? p.toLowerCase() : p;
}

export function relative( id: string, cwd: string ): string {
  return slash( lowerPath( path.relative( cwd, id ) ) );
}

export function relativeURL( id: string, cwd: string ): string {
  return slash( lowerPath( path.relative( path.dirname( cwd ), id ) ) );
}

export function resolvePath( id: string, cwd: string ): string {
  return lowerPath( path.resolve( cwd, id ) );
}

export function getType( id: string ): string {
  const match = id.match( reExt );
  return match ? match[ 1 ] : "";
}

export function makeAbsolute( file: string ) {
  return lowerPath( path.resolve( process.cwd(), file ) );
}

export function isAbsolute( file: string ): boolean {
  return path.isAbsolute( file );
}
