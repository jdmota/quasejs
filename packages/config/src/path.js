// @flow
import { formatOption } from "./formating";

export default class Path {

  +arr: string[];
  where: ?string;

  constructor() {
    this.arr = [];
    this.where = null;
  }

  push( key: string | number ) {
    this.arr.push( key + "" );
  }

  pop() {
    this.arr.pop();
  }

  size() {
    return this.arr.length;
  }

  chainToString() {
    return this.arr.join( "." );
  }

  format() {
    return formatOption( `${this.arr.join( "." )}${this.where ? ` [in ${this.where}]` : ""}` );
  }

}
