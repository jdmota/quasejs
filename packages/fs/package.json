{
  "private": true,
  "name": "@quase/fs",
  "version": "0.1.0",
  "description": "FS",
  "license": "MIT",
  "homepage": "https://github.com/quase/quasejs",
  "repository": "https://github.com/quase/quasejs/tree/master/packages/fs",
  "main": "dist/index.js",
  "module": "src/index.js",
  "files": [
    "dist/index.js",
    "dist/index.js.map",
    "src"
  ],
  "dependencies": {
    "fs-extra": "^4.0.2",
    "lru-cache": "^4.1.1",
    "micromatch": "^2.3.11",
    "slash": "^1.0.0",
    "zen-observable": "^0.5.2"
  }
}
