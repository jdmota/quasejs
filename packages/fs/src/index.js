const fs = require( "fs-extra" );

// TODO add https://github.com/zkochan/symlink-dir
// TODO add https://github.com/IndigoUnited/node-proper-lockfile

export const {
  FSWatcher,
  ReadStream,
  Stats,
  WriteStream,
  constants,
  createReadStream,
  createWriteStream,
  unwatchFile,
  watch,

  access,
  appendFile,
  chmod,
  chown,
  close,
  exists,
  fchmod,
  fchown,
  fdatasync,
  fstat,
  fsync,
  ftruncate,
  futimes,
  lchmod,
  lchown,
  link,
  lstat,
  mkdir,
  mkdtemp,
  open,
  read,
  readdir,
  readFile,
  readlink,
  realpath,
  rename,
  rmdir,
  stat,
  symlink,
  truncate,
  unlink,
  utimes,
  write,
  writeFile,

  copy,
  emptyDir,
  ensureDir,
  ensureFile,
  ensureLink,
  ensureSymlink,
  move,
  outputFile,
  outputJson,
  readJson,
  remove,
  writeJson
} = fs;

// Alias
export {
  emptyDir as emptydir,
  ensureDir as mkdirs,
  ensureDir as mkdirp,
  ensureFile as createFile,
  readJson as readJSON,
  writeJson as writeJSON
};
