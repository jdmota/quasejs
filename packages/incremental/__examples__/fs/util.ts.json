{
  "content": "content of util.ts",
  "deps": []
}
