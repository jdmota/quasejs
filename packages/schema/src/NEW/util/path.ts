export class Path {
  public readonly parent: Path | null;
  public readonly key: string | number | null;
  public readonly context: string | null;

  constructor(
    parent: Path,
    key: string | number | null,
    context: string | null
  );
  constructor(parent: null, key: null, context: null);
  constructor(
    parent: Path | null,
    key: string | number | null,
    context: string | null
  ) {
    this.parent = parent;
    this.key = key;
    this.context = context;
  }

  isRoot() {
    return this.parent == null;
  }

  push(key: string | number | null, context: string | null = null) {
    return new Path(this, key, context);
  }

  pop(): Path {
    if (this.parent == null)
      throw new Error("Invariant: validation path - cannot pop the root");
    return this.parent;
  }

  static create() {
    return new Path(null, null, null);
  }
}
