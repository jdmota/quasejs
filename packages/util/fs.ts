import fs from "fs/promises";

// https://nodejs.org/api/errors.html#nodejs-error-codes

// "No such file or directory"
export function isNoFileError(error: any) {
  return error != null && error.code === "ENOENT";
}

export async function readFileString(filename: string): Promise<string> {
  try {
    return await fs.readFile(filename, "utf8");
  } catch (error) {
    if (isNoFileError(error)) {
      return "";
    }
    throw error;
  }
}
