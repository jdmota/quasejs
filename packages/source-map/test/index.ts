import fs from "fs-extra";
import path from "path";
import { fileURLToPath } from "url";
import {
  joinSourceMaps,
  SourceMapExtractor,
  SourceMapExtractorBase,
} from "../src";

const dirname = path.dirname(fileURLToPath(import.meta.url));

function relative(a: string, b: string) {
  return path.relative(a, b).replace(/\\/g, "/");
}

test("get map from file content", async () => {
  const extractor = new SourceMapExtractorBase();
  const fixturesFolder = path.join(dirname, "fixtures");
  const files = await fs.readdir(fixturesFolder);

  for (const file of files) {
    if (/\.map$/.test(file)) {
      continue;
    }
    const fileLocation = path.join(dirname, "fixtures", file);
    const info = extractor.getMapFromFile(
      fileLocation,
      await fs.readFile(fileLocation, "utf8")
    );

    info!.mapLocation = relative(fixturesFolder, info!.mapLocation);
    expect(info).toMatchSnapshot();
  }
});

test("get map from file", async () => {
  const extractor = new SourceMapExtractor();
  const fixturesFolder = path.join(dirname, "fixtures");
  const files = await fs.readdir(fixturesFolder);

  for (const file of files) {
    if (/\.map$/.test(file)) {
      continue;
    }
    const fileLocation = path.join(dirname, "fixtures", file);
    const info = await extractor.getMap(fileLocation);

    info!.mapLocation = relative(fixturesFolder, info!.mapLocation);
    expect(info).toMatchSnapshot();
  }
});

test("get original location from map", async () => {
  const extractor = new SourceMapExtractorBase();
  const fixturesFolder = path.join(dirname, "fixtures");
  const mapLocation = path.join(fixturesFolder, "map-file-comment.css.map");
  const map = JSON.parse(await fs.readFile(mapLocation, "utf8"));

  const location = await extractor.getOriginalLocationFromMap(
    map,
    mapLocation,
    {
      line: 8,
      column: 0,
    }
  );

  location!.originalFile = relative(fixturesFolder, location!.originalFile!);
  expect(location).toMatchSnapshot();
});

test("joinSourceMaps", async () => {
  // TODO better test

  const fixturesFolder = path.join(dirname, "fixtures");
  const mapLocation = path.join(fixturesFolder, "map-file-comment.css.map");
  const map = JSON.parse(await fs.readFile(mapLocation, "utf8"));

  expect(await joinSourceMaps([map, map, map])).toMatchSnapshot();
});
