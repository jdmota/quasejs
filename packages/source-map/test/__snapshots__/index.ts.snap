// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get map from file 1`] = `
{
  "map": {
    "file": "map-file-comment.css",
    "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
    "sources": [
      "./client/sass/core.scss",
      "./client/sass/main.scss",
    ],
    "version": "3",
  },
  "mapLocation": "multi-line-inline.css",
}
`;

exports[`get map from file 2`] = `
{
  "map": {
    "file": "map-file-comment.css",
    "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
    "sources": [
      "./client/sass/core.scss",
      "./client/sass/main.scss",
    ],
    "version": "3",
  },
  "mapLocation": "map-file-comment.css.map",
}
`;

exports[`get map from file 3`] = `
{
  "map": {
    "file": "map-file-comment.css",
    "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
    "sources": [
      "./client/sass/core.scss",
      "./client/sass/main.scss",
    ],
    "version": "3",
  },
  "mapLocation": "single-line-inline.css",
}
`;

exports[`get map from file 4`] = `
{
  "map": {
    "file": "map-file-comment.css",
    "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
    "sources": [
      "./client/sass/core.scss",
      "./client/sass/main.scss",
    ],
    "version": "3",
  },
  "mapLocation": "map-file-comment.css.map",
}
`;

exports[`get map from file content 1`] = `
{
  "map": {
    "file": "map-file-comment.css",
    "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
    "sources": [
      "./client/sass/core.scss",
      "./client/sass/main.scss",
    ],
    "version": "3",
  },
  "mapLocation": "multi-line-inline.css",
}
`;

exports[`get map from file content 2`] = `
{
  "map": null,
  "mapLocation": "map-file-comment.css.map",
}
`;

exports[`get map from file content 3`] = `
{
  "map": {
    "file": "map-file-comment.css",
    "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
    "sources": [
      "./client/sass/core.scss",
      "./client/sass/main.scss",
    ],
    "version": "3",
  },
  "mapLocation": "single-line-inline.css",
}
`;

exports[`get map from file content 4`] = `
{
  "map": null,
  "mapLocation": "map-file-comment.css.map",
}
`;

exports[`get original location from map 1`] = `
{
  "column": 0,
  "line": 9,
  "name": null,
  "originalCode": null,
  "originalFile": "client/sass/core.scss",
  "source": "client/sass/core.scss",
}
`;

exports[`joinSourceMaps 1`] = `
{
  "file": "map-file-comment.css",
  "mappings": "AAAA,wBAAyB;EACvB,UAAU,EAAE,IAAI;EAChB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,iBAAiB;EAChC,KAAK,EAAE,OAAkB;;AAG3B,wBAAyB;EACvB,OAAO,EAAE,IAAI;;ACTf,gBAAiB;EACf,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,MAAM;;AAGf,kBAAmB;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,KAAK;EACjB,aAAa,EAAE,GAAG;EAClB,KAAK,EAAE,KAAK;;AAEd,kBAAmB;EACjB,KAAK,EAAE,KAAK;;AAGd,mBAAoB;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,IAAI",
  "names": [],
  "sources": [
    "client/sass/core.scss",
    "client/sass/main.scss",
  ],
  "version": 3,
}
`;
